### PR TITLE
feat: Add shortcut creation

### PR DESCRIPTION
**Pull request by Crash--**
Add a modal to create a shortcut from the More Menu

Capture d’écran 2020-03-16 à 11 45 09

### CODE REVIEW

File: src/drive/locales/en.json [hunk 1]
```diff
@@ -643,5 +644,16 @@
...
+  "Shortcut": {
+    "title_modal": "Create a shortcut",
+    "content_modal": "You can create a shortcut",
+    "filename": "Filename",
+    "url": "URL",
+    "cancel": "Cancel",
+    "create": "Create",
+    "created": "Your shortcut has been created",
+    "errored": "An error occured",
+    "filename_error_ends": "The name should ends up with .url"
```
Comment (y-lohse) [inline src/drive/locales/en.json]:
```suggestion
    "filename_error_ends": "The name should end with .url"
```

File: src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const CreateShortcutWrapper = () => {
+  const [isModalDisplayed, setIsModalDisplayed] = useState(false)
+  const { t } = useI18n()
+  if (!isModalDisplayed) {
+    return (
+      <a
+        data-test-id="create-a-note"
+        className={styles['fil-action-create-shortcut']}
+        onClick={() => {
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx]: We can use `useCallback` here

File: src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          setIsModalDisplayed(true)
+        }}
+      >
+        {t('toolbar.menu_create_shortcut')}
+      </a>
+    )
+  } else {
+    return <ShortcutCreationModal onClose={() => setIsModalDisplayed(false)} />
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx]: Here as well :)

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    try {
+      await client.collection('io.cozy.files.shortcuts').create(data)
+      Alerter.success('Shortcut.created')
+      onClose()
+    } catch (e) {
+      Alerter.error('Shortcut.errored')
+    }
+  }
+  return (
+    <ExperimentalDialog open={true} onClose={() => onClose()}>
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]: I don't think we need the explicit `open` here?

Comment (y-lohse) [reply src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]:
```suggestion
    <ExperimentalDialog open={true} onClose={onClose}>
```

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      <DialogCloseButton onClick={() => onClose()} />
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]:
```suggestion
      <DialogCloseButton onClick={onClose} />
```

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      </ExperimentalDialogTitle>
+      <Divider />
+      <DialogContent>
+        <Stack>
+          <Label htmlFor="shortcuturl">{t('Shortcut.url')}</Label>
+          <Input
+            id="shortcuturl"
+            fullwidth
+            onChange={e => setUrl(e.target.value)}
+          />
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]: Is this according to the zeplin specs? I think new designs were supposed to use https://docs.cozy.io/cozy-ui/react/#/Material-UI?id=textfield

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+              onChange={e => setFilename(e.target.value)}
+            />
+          </InputGroup>
+        </Stack>
+      </DialogContent>
+
+      <ExperimentalDialogActions>
+        <Button
+          theme="secondary"
+          onClick={() => onClose()}
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]:
```suggestion
          onClick={onClose}
```

File: src/drive/locales/en.json [hunk 8]
```diff
@@ -643,5 +644,16 @@
       "text": "You're about to be redirected…",
       "error": "Error during the redirection. Generally, this means that the content of the file is not in the correct format."
     }
+  },
+  "Shortcut": {
+    "title_modal": "Create a shortcut",
+    "content_modal": "You can create a shortcut",
```
Comment (y-lohse) [inline src/drive/locales/en.json]: I don't think this string is used, is it?

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+          theme="primary"
+          label={t('Shortcut.create')}
+          onClick={useCallback(createShortcut)}
+        />
+      </ExperimentalDialogActions>
+    </ExperimentalDialog>
+  )
+}
+
+export default toolbarContainer(ShortcutCreationModal)
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]: Would it be difficult to test this modal through react-testing-lib ? Filling in the form fiels and checking that we make a request to the stack?

File: src/drive/locales/en.json (hunk 8 above)
Comment (Crash--) [reply src/drive/locales/en.json]: You right, we removed it. Thanks!

File: src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx (hunk 9 above)
Comment (Crash--) [reply src/drive/web/modules/drive/Toolbar/components/ShortcutCreationModal.jsx]: Let's try. Thanks

File: src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+  const [isModalDisplayed, setIsModalDisplayed] = useState(false)
+  const { t } = useI18n()
+  if (!isModalDisplayed) {
+    return (
+      <a
+        data-test-id="create-a-shortcut"
+        className={styles['fil-action-create-shortcut']}
+        onClick={useCallback(() => {
+          setIsModalDisplayed(true)
+        })}
```
Comment (y-lohse) [inline src/drive/web/modules/drive/Toolbar/components/CreateShortcut.jsx]: I don't mind if it's not in the PR, but I was more thinking of something along the lines of:

```
const toggleModal = useCallback(() => setIsModalDisplayed(! isModalDisplayed))
…
<a onClick={toggleModal}>
…
<ShortcutCreationModal onClose={toggleModal} />
```

(I think hooks are really elegant for this sort of things)

**Pull request merged**